py3status/modules/mpd_status.py

```diff
@@ -167,12 +180,14 @@ def attr_getter(attr):
         except socket.error:
             text = "Failed to connect to mpd!"
             state = None
+        except ConnectionError:
+            text = "Error while connecting to mpd!"
+            state = None
+            self._get_mpd(disconnect=True)
         except CommandError:
             text = "Failed to authenticate to mpd!"
             state = None
-            c.disconnect()
-        else:
-            c.disconnect()
+            self._get_mpd(disconnect=True)
```

Comment:
Everything looks okay here here except this... I can cause `CommandError` here with successful socket then broken socket... and playing with auth passwords. I know we made `get_mpd` by taking some code out of try/except here... and making identical try/except here too. 

@tobes Suggestions? I used `self.py3.error('incorrect password')` here wanting to stop looping immediately... I get that on the bar okay... but still see `mpd_status` running in the log... New client, new password, new attempt, fail... All of that kept going every few seconds.

Reply:
well spotted! I'd suggest wrapping the disconnect part in a try block (without catching anything)

Reply:
I think most of the users don't use password anyway... And this is a bug that had nothing to do with this PR... Instead of try block, I think we can fix `get_mpd()` to check `self.client` first before doing anything... Another thing to consider... Place `if disconnect` under `get_mpd()`'s try block.

Reply:
I'd do: 

        if disconnect:
            try:
                self.client.disconnect()
            finally:
                self.client = None
            return

for now, and maybe open a new issue about that?

I wouldn't place `if disconnect` under the other try-catch block, because throwing an exception on disconnect is unnecessary. just ignore it.

Reply:
It's fine either way. That exception is extremely rare. I'd be amazed if the users even can trip this as they would have to keep mangling with the password/sockets repeatedly like I did to catch that.

The owners/collaborators are currently too busy to review anything. I would not want to open an issue as I think this is trivial and we can fix it here now..  and wait for their return.